File: 2015/01/01.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, err := os.ReadFile("2015/01/01-input.txt")
	if err != nil {
		panic(err)
	}
	directions := string(file)

	// Count the number of open and close parentheses
	var floor int
	var basement int
	for i, direction := range directions {
		if direction == '(' {
			floor++
		} else {
			floor--
			// Track the first time direction leads to basement
			if floor == -1 && basement == 0 {
				basement = i + 1
			}
		}
	}

	// Part One: The floor Santa ends up on
	fmt.Printf("Part One: %d\n", floor)

	// Part Two: The position of the character that causes Santa to first enter the basement
	fmt.Printf("Part Two: %d\n", basement)
}

File: 2015/01/01.py
# Read input as string
with open("01_input.txt", 'r') as file:
    directions = file.read()

# Count the number of open and close parentheses
floor = 0
basement = 0
for i, direction in enumerate(directions):
    if direction == '(':
        floor += 1
    else:
        floor -= 1
        # Track the first time direction leads to basement
        if floor == -1 and basement == 0:
            basement = i + 1

# Part One: The floor Santa ends up on
print("Part One:", floor)

# Part Two: The position of the character that causes Santa to first enter the basement
print("Part Two:", basement)
